From 940a3e67453a4908a7d37b2e69694a22108f55d4 Mon Sep 17 00:00:00 2001
From: Prarit Bhargava <prarit@redhat.com>
Date: Fri, 19 Sep 2014 12:29:23 -0400
Subject: [acpi] Drop acpi_evaluate_hotplug_ost() and ACPI_HOTPLUG_OST

Message-id: <1411129776-21656-49-git-send-email-prarit@redhat.com>
Patchwork-id: 96380
O-Subject: [RHEL7.1 PATCH BZ 1141846 48/61] ACPI: Drop acpi_evaluate_hotplug_ost() and ACPI_HOTPLUG_OST
Bugzilla: 1141846
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Myron Stowe <mstowe@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1141846

commit 700b8422fb9e6041907bda45a15f4e2ab0438a04
Author: Rafael J. Wysocki <rafael.j.wysocki@intel.com>
Date:   Fri Feb 21 01:07:17 2014 +0100

    ACPI: Drop acpi_evaluate_hotplug_ost() and ACPI_HOTPLUG_OST

    Replace acpi_evaluate_hotplug_ost() with acpi_evaluate_ost()
    everywhere and drop the ACPI_HOTPLUG_OST symbol so that hotplug
    _OST is supported unconditionally.

    Signed-off-by: Rafael J. Wysocki <rafael.j.wysocki@intel.com>
    Acked-by: Toshi Kani <toshi.kani@hp.com>

Cc: Lenny Szubowicz <lszubowi@redhat.com>
Cc: Stefan Assmann <sassmann@redhat.com>
Cc: Tony Camuso <tcamuso@redhat.com>
Cc: Vitaly Kuznetsov <vkuznets@redhat.com>
Cc: Radim Krcmar <rkrcmar@redhat.com>
Cc: Igor Mammedov <imammedo@redhat.com>
Cc: Andrew Jones <drjones@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/acpi/bus.c b/drivers/acpi/bus.c
index cb7c930..9fa67c7 100644
--- a/drivers/acpi/bus.c
+++ b/drivers/acpi/bus.c
@@ -312,9 +312,7 @@ static void acpi_bus_osc_support(void)
  capbuf[OSC_SUPPORT_DWORD] |= OSC_SB_PPC_OST_SUPPORT;
 #endif
 
-#ifdef ACPI_HOTPLUG_OST
  capbuf[OSC_SUPPORT_DWORD] |= OSC_SB_HOTPLUG_OST_SUPPORT;
-#endif
 
  if (!ghes_disable)
   capbuf[OSC_SUPPORT_DWORD] |= OSC_SB_APEI_SUPPORT;
@@ -509,7 +507,7 @@ static void acpi_bus_notify(acpi_handle handle, u32 type, void *data)
  return;
 
  err:
- acpi_evaluate_hotplug_ost(handle, type, ost_code, NULL);
+ acpi_evaluate_ost(handle, type, ost_code, NULL);
 }
 
 /* --------------------------------------------------------------------------
diff --git a/drivers/acpi/scan.c b/drivers/acpi/scan.c
index 40f2fdc..15f2647 100644
--- a/drivers/acpi/scan.c
+++ b/drivers/acpi/scan.c
@@ -404,8 +404,8 @@ static int acpi_generic_hotplug_event(struct acpi_device *adev, u32 type)
    dev_info(&adev->dev, "Eject disabled\n");
    return -EPERM;
   }
-  acpi_evaluate_hotplug_ost(adev->handle, ACPI_NOTIFY_EJECT_REQUEST,
-       ACPI_OST_SC_EJECT_IN_PROGRESS, NULL);
+  acpi_evaluate_ost(adev->handle, ACPI_NOTIFY_EJECT_REQUEST,
+      ACPI_OST_SC_EJECT_IN_PROGRESS, NULL);
   return acpi_scan_hot_remove(adev);
  }
  return -EINVAL;
@@ -455,7 +455,7 @@ void acpi_device_hotplug(void *data, u32 src)
   ost_code = ACPI_OST_SC_SUCCESS;
 
  err_out:
- acpi_evaluate_hotplug_ost(adev->handle, src, ost_code, NULL);
+ acpi_evaluate_ost(adev->handle, src, ost_code, NULL);
 
  out:
  acpi_bus_put_acpi_device(adev);
@@ -515,8 +515,8 @@ acpi_eject_store(struct device *d, struct device_attribute *attr,
   return count;
 
  put_device(&acpi_device->dev);
- acpi_evaluate_hotplug_ost(acpi_device->handle, ACPI_OST_EC_OSPM_EJECT,
-      ACPI_OST_SC_NON_SPECIFIC_FAILURE, NULL);
+ acpi_evaluate_ost(acpi_device->handle, ACPI_OST_EC_OSPM_EJECT,
+     ACPI_OST_SC_NON_SPECIFIC_FAILURE, NULL);
  return status == AE_NO_MEMORY ? -ENOMEM : -EAGAIN;
 }
 
diff --git a/drivers/acpi/utils.c b/drivers/acpi/utils.c
index e2edf0d..0d818ed 100644
--- a/drivers/acpi/utils.c
+++ b/drivers/acpi/utils.c
@@ -419,7 +419,7 @@ out:
 EXPORT_SYMBOL(acpi_get_physical_device_location);
 
 /**
- * acpi_evaluate_hotplug_ost: Evaluate _OST for hotplug operations
+ * acpi_evaluate_ost: Evaluate _OST for hotplug operations
  * @handle: ACPI device handle
  * @source_event: source event code
  * @status_code: status code
diff --git a/drivers/xen/xen-acpi-cpuhotplug.c b/drivers/xen/xen-acpi-cpuhotplug.c
index 73496c3..410c6d5 100644
--- a/drivers/xen/xen-acpi-cpuhotplug.c
+++ b/drivers/xen/xen-acpi-cpuhotplug.c
@@ -316,7 +316,7 @@ static void acpi_processor_hotplug_notify(acpi_handle handle,
   goto out;
  }
 
- (void) acpi_evaluate_hotplug_ost(handle, event, ost_code, NULL);
+ (void) acpi_evaluate_ost(handle, event, ost_code, NULL);
 
 out:
  acpi_scan_lock_release();
diff --git a/drivers/xen/xen-acpi-memhotplug.c b/drivers/xen/xen-acpi-memhotplug.c
index 9b056f0..182262c 100644
--- a/drivers/xen/xen-acpi-memhotplug.c
+++ b/drivers/xen/xen-acpi-memhotplug.c
@@ -286,7 +286,7 @@ static void acpi_memory_device_notify(acpi_handle handle, u32 event, void *data)
   return;
  }
 
- (void) acpi_evaluate_hotplug_ost(handle, event, ost_code, NULL);
+ (void) acpi_evaluate_ost(handle, event, ost_code, NULL);
  return;
 }
 
diff --git a/include/acpi/acpi_bus.h b/include/acpi/acpi_bus.h
index dcdcd80..989538c 100644
--- a/include/acpi/acpi_bus.h
+++ b/include/acpi/acpi_bus.h
@@ -53,14 +53,6 @@ acpi_evaluate_reference(acpi_handle handle,
 acpi_status
 acpi_evaluate_ost(acpi_handle handle, u32 source_event, u32 status_code,
     struct acpi_buffer *status_buf);
-#ifdef ACPI_HOTPLUG_OST
-#define acpi_evaluate_hotplug_ost acpi_evaluate_ost
-#else
-static inline acpi_status
-acpi_evaluate_hotplug_ost(acpi_handle handle, u32 source_event,
-   u32 status_code, struct acpi_buffer *status_buf)
-{ return AE_OK; }
-#endif
 
 acpi_status
 acpi_get_physical_device_location(acpi_handle handle, struct acpi_pld_info **pld);
-- 
1.7.1